persistence/trees/module.go

```diff
@@ -41,18 +43,21 @@ func WithLogger(logger *modules.Logger) modules.TreeStoreOption {
 // saves its data.
 func WithTreeStoreDirectory(path string) modules.TreeStoreOption {
 	return func(m modules.TreeStoreModule) {
-		if mod, ok := m.(*treeStore); ok {
-			mod.treeStoreDir = path
+		mod, ok := m.(*TreeStore)
+		if ok {
+			mod.TreeStoreDir = path
```

Comment:
```suggestion
			mod.treeStoreDir = path
```
Does this need to be exported? We will always know what it's value is as we pass it in ourselves on creation